crates/types/src/consensus.rs

```diff
@@ -299,7 +299,27 @@ impl<TYPES: NodeType> Consensus<TYPES> {
     }
 
     /// Update the validated state map with a new view_number/view combo.
+    ///
+    /// Skips the update if the new view contains less information than the exisiting view with the
+    /// same view number.
     pub fn update_validated_state_map(&mut self, view_number: TYPES::Time, view: View<TYPES>) {
+        if let Some(existing_view) = self.validated_state_map().get(&view_number) {
+            if let ViewInner::Leaf { .. } = existing_view.view_inner {
+                match view.view_inner.clone() {
```

Comment:
Do we need to clone here?

Reply:
I added this to fix a build error. Let me double-check!

Reply:
Yeah looks like it's needed. It's due to a partial value move a few lines later on `ViewInner::Leaf { delta, .. }`.